registry/api/v2/descriptors.go

```diff
@@ -1595,6 +1595,133 @@ var routeDescriptors = []RouteDescriptor{
 			},
 		},
 	},
+	{
+		Name:        RouteNameReferrers,
+		Path:        "/v2/{name:" + reference.NameRegexp.String() + "}/referrers/{digest:" + digest.DigestRegexp.String() + "}",
+		Entity:      "Referrers",
+		Description: `Retrieve information about referrers.`,
+		Methods: []MethodDescriptor{
+			{
+				Method:      "GET",
+				Description: "Fetch the referrers of the artifact identified by `digest`.",
+				Requests: []RequestDescriptor{
+					{
+						Name:        "referrers",
+						Description: "Request an unabridged list of referrers.",
+						Successes: []ResponseDescriptor{
+							{
+								Description: "Returns an image index containing all referrers as a json response.",
+								StatusCode:  http.StatusOK,
+								Headers: []ParameterDescriptor{
+									{
+										Name:        "Content-Length",
+										Type:        "integer",
+										Description: "Length of the JSON response body.",
+										Format:      "<length>",
+									},
+									linkHeader,
+								},
+								Body: BodyDescriptor{
+									ContentType: "application/json",
+									Format: `{
+												"schemaVersion": 2,
+												"mediaType": "application/vnd.oci.image.index.v1+json",
+												"manifests": [
+													<manifest>,
+													...
+												]
+											}`,
+								},
+							},
+						},
+						Failures: []ResponseDescriptor{
+							{
+								Description: "The registry does not support referrers API.",
+								StatusCode:  http.StatusNotFound,
+							},
```

Comment:
It is also possible that the repository does not exists.

/cc @Wwwsylvia @m5i-work

Reply:
We should return 404 and error code `NAME_UNKNOWN` when the repository does not exist.
References:
- https://github.com/opencontainers/distribution-spec/blob/v1.1.0-rc1/spec.md#error-codes
- https://docs.docker.com/registry/spec/api/#errors-2

Reply:
Added the variable `repositoryNotFoundResponseDescriptor` defined in `descriptors.go` to address this scenario. It satisfies your specifications.
<img width="488" alt="image" src="https://user-images.githubusercontent.com/103478229/199933018-be3843fa-ad24-4f68-acee-111aaf3565cc.png">
Now the Failures responses look like this, note the added line:
<img width="1170" alt="image" src="https://user-images.githubusercontent.com/103478229/199933652-2c300d19-4f84-4233-82d8-59806f847617.png">